### ISSUE TITLE
be crash

### ISSUE THREAD

**Issue by wanpengfei-git** [StarRocks/starrocks#4955]
### Steps to reproduce the behavior (Required)
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
*** Aborted at 1649610624 (unix time) try "date -d @1649610624" if you are using GNU date ***
PC: @     0x7fc7595d4387 __GI_raise
*** SIGABRT (@0x3e80000629c) received by PID 25244 (TID 0x7fc503e23700) from PID 25244; stack trace: ***
    @     0x7fc75a089630 (unknown)
    @     0x7fc7595d4387 __GI_raise
    @     0x7fc7595d5a78 __GI_abort
    @          0x187e371 _ZN9__gnu_cxx27__verbose_terminate_handlerEv.cold
    @          0x57027e6 __cxxabiv1::__terminate()
    @          0x5702851 std::terminate()
    @          0x57029a4 __cxa_throw
    @          0x187e278 _Znwm.cold
    @          0x208ef76 std::vector<>::_M_default_append()
    @          0x208b224 starrocks::BackendService_submit_tasks_args::read()
    @          0x208b2d2 starrocks::BackendServiceProcessor::process_submit_tasks()
    @          0x208aa52 starrocks::BackendServiceProcessor::dispatchCall()
    @          0x208c732 apache::thrift::TDispatchProcessor::process()
    @          0x3d7fa35 apache::thrift::server::TConnectedClient::run()
    @          0x3d75e33 apache::thrift::server::TThreadedServer::TConnectedClientRunner::run()
    @          0x3d79d1f apache::thrift::concurrency::Thread::threadMain()
    @          0x3d58e67 std::__invoke_impl<>()
    @          0x3d58dbb std::__invoke<>()
    @          0x3d58d2b _ZNSt6thread8_InvokerISt5tupleIJPFvSt10shared_ptrIN6apache6thrift11concurrency6ThreadEEES7_EEE9_M_invokeIJLm0ELm1EEEEvSt12_Index_tupleIJXspT_EEE
    @          0x3d58c7c std::thread::_Invoker<>::operator()()
    @          0x3d58be0 std::thread::_State_impl<>::_M_run()
    @          0x577cbc0 execute_native_thread_routine
    @     0x7fc75a081ea5 start_thread
    @     0x7fc75969c8dd __clone
    @                0x0 (unknown)

```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - branch-2.2 Release
 - mysql> select current_version();
+---------------------------------+
| current_version()               |
+---------------------------------+
| SQLANCER_02_RELEASE 790f3a4 |
+---------------------------------+
- path: daily05

**Comment by chaoyli** [comment]
There are 500 thousand tasks in ASAN mode. It consumes too much memory and throws a std::bad_alloc exception.

**Comment by wangruin** [comment]
https://github.com/StarRocks/starrocks/pull/5781
This PR has been merged into branch-2.2, you can see if there is still crash.
